hal/common/mbed_alloc_wrappers.c

```diff
@@ -0,0 +1,102 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed_mem_trace.h"
+#include <stddef.h>
+#include <stdio.h>
+
+#if defined(TOOLCHAIN_GCC)
+
+#ifdef   FEATURE_UVISOR
+#include "uvisor-lib/uvisor-lib.h"
+#endif/* FEATURE_UVISOR */
+
+// TODO: memory tracing doesn't work with uVisor enabled.
+#if !defined(FEATURE_UVISOR)
+void * __wrap__malloc_r(struct _reent * r, size_t size) {
+    extern void * __real__malloc_r(struct _reent * r, size_t size);
+#ifdef MBED_MEM_TRACING_ENABLED
```

Comment:
Via config, yes, but not necessarily a configuration parameter. See https://github.com/ARMmbed/mbed-os/pull/2442/files#r75299576 for more details.